cpp/src/strings/json/json_to_array.cu

```diff
@@ -0,0 +1,563 @@
+#include <cudf/column/column_factories.hpp>
+#include <cudf/detail/get_value.cuh>
+#include <cudf/detail/utilities/cuda.cuh>
+#include <cudf/scalar/scalar.hpp>
+#include <cudf/scalar/scalar_factories.hpp>
+#include <cudf/strings/strings_column_view.hpp>
+#include <cudf/strings/detail/utilities.hpp>
+#include <cudf/types.hpp>
+#include <cudf/utilities/error.hpp>
+#include <cudf/structs/structs_column_view.hpp>
+
+#include <rmm/exec_policy.hpp>
+
+namespace cudf {
+
+class parser {
+  protected:
+  CUDA_HOST_DEVICE_CALLABLE parser() : input(nullptr), input_len(0), pos(nullptr) {}
+  CUDA_HOST_DEVICE_CALLABLE parser(const char* _input, int64_t _input_len)
+    : input(_input), input_len(_input_len), pos(_input)
+  {
+    parse_whitespace();
+  }
+  CUDA_HOST_DEVICE_CALLABLE bool parse_whitespace()
+  {
+    while (!eof()) {
+      char c = *pos;
+      if (c == ' ' || c == '\r' || c == '\n' || c == '\t') {
+        pos++;
+      } else {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  CUDA_HOST_DEVICE_CALLABLE bool eof(const char* p) { return p - input >= input_len; }
+
+  CUDA_HOST_DEVICE_CALLABLE bool eof() { return eof(pos); }
+
+  protected:
+  char const* input;
+  int64_t input_len;
+  char const* pos;
+};
+
+CUDA_HOST_DEVICE_CALLABLE bool device_strncmp(const char* str1, const char* str2, size_t num_chars)
+{
+  for (size_t idx = 0; idx < num_chars; idx++) {
+    if (str1[idx] != str2[idx]) { return false; }
+  }
+  return true;
+}
+
+
+CUDA_HOST_DEVICE_CALLABLE bool is_digit(char c)
+{
+  if (c >= '0' && c <= '9') return true;
+  return false;
+}
+
+struct json_string {
+  const char* str;
+  int64_t len;
```

Comment:
Can a string ever be this big?

Reply:
I assume that the entire json file is loaded as one string, thus it can be that big (TB in size).